packages/cli/src/GenericHelpers.ts

```diff
@@ -39,7 +39,7 @@ export async function validateEntity(
 export const DEFAULT_EXECUTIONS_GET_ALL_LIMIT = 20;
 
 class StringWithNoXss {
-	@NoXss()
+	@NoXss({ each: true })
 	value: string;
```

Comment:
The issue here is that it now assumes that `value` is `string[]`. Before it assumed it was `string`. However the `validateRecordNoXss` receives an object, which can contain anything (it assumes it's a `Record<string, string>`). We need to check the type of each property in the object and apply the correct validator based on that.

Only `string`, `array` or `object` can be a possible Xss, so we need to handle all those different types.

Something like

```ts
if (typeof value === "string") { validateString }
else if (Array.isArray(value)) { validateArray }
else if (isObject(value)) { validateAllObjectsPropertiesAndNestedProperties }
```

Since this is only used for the personalization survey, it starts to sound to me that it would be much easier to define the expected Payload dto and specify the validators for that, instead of trying to handle the very generic case